cmd/internal/kube/secret.go

```
@@ -0,0 +1,127 @@
+
+		err = handleSecretGeneration(namespace, key, secret, output)
+		if err != nil {
+			return err
+		}
+
+		printer.Infoln("Generated Kubernetes secret config to ", output)
+		return nil
+	},
+}
+
+// Represents the input used by secretTemplate
+type secretTemplateInput struct {
+	Namespace string
+	APIKey    string
+	APISecret string
+}
+
+// Generates a Kubernetes secret config file for Akita
+func handleSecretGeneration(namespace, key, secret, output string) error {
+
+	input := secretTemplateInput{
+		Namespace: namespace,
+		APIKey:    base64.StdEncoding.EncodeToString([]byte(key)),
+		APISecret: base64.StdEncoding.EncodeToString([]byte(secret)),
+	}
+
+	secretFile, err := createSecretFile(output)
+	if err != nil {
+		return cmderr.AkitaErr{Err: errors.Wrap(err, "failed to create output file")}
+	}
+
+	defer secretFile.Close()
+
+	err = secretTemplate.Execute(secretFile, input)
+	if err != nil {
+		return cmderr.AkitaErr{Err: errors.Wrap(err, "failed to generate template")}
+	}
+
+	return nil
+}
+
+// Creates a file at the give path to be used for storing of the generated Secret config
+// If any child dicrectories do not exist, it will be created.
+func createSecretFile(path string) (*os.File, error) {
+	// Split the outut flag value into directory and filename
+	outputDir, outputName := filepath.Split(path)
+
+	// Get the absolute path of the output directory
+	absOutputDir, err := filepath.Abs(outputDir)
+	if err != nil {
+		return nil, errors.Wrapf(err, "failed to resolve the absolute path of the output directory")
+	}
+
+	// Check if the output file already exists
+	outputFilePath := filepath.Join(absOutputDir, outputName)
+	if _, statErr := os.Stat(outputFilePath); statErr == nil {
+		return nil, errors.Errorf("output file %s already exists", outputFilePath)
+	}
+
+	// Create the output directory if it doesn't exist
+	err = os.MkdirAll(absOutputDir, 0755)
+	if err != nil {
+		return nil, errors.Wrapf(err, "failed to create the output directory")
+	}
```

It seems a little unusual to me for a tool of this type to create the directory.  Probably I would just raise an error at this point.